common/script/ops/purchase.js

```diff
@@ -3,105 +3,127 @@ import i18n from '../i18n';
 import _ from 'lodash';
 import splitWhitespace from '../libs/splitWhitespace';
 import planGemLimits from '../libs/planGemLimits';
+import {
+  NotFound,
+  NotAuthorized,
+  BadRequest,
+} from '../libs/errors';
+
+module.exports = function purchase (user, req = {}, analytics) {
+  let parameters = req.body;
```

Comment:
parameters are passed in req.parameter not in req.body, we can't change it (I'd like too to use req.body) because this code is used for v2 API too. Where possible in shared code I'd use `_.get(req.parameters.key)` so if parameters doesn't exists it doesn't break everything.

You can see how parameters are passed here https://github.com/HabitRPG/habitrpg/blob/develop/website/src/routes/api-v2/swagger.js#L162 And the same url format should also be used for https://github.com/HabitRPG/habitrpg/pull/6988/files#diff-9b5af26375c670974e1a8e7acd33569bR687